xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -308,58 +311,7 @@ bool CPVRClient::GetAddonProperties()
 
   if (retVal != PVR_ERROR_NO_ERROR)
     return false;
-
-  /* get the name of the backend */
-  retVal = DoAddonCall(
-      __func__,
-      [&strBackendName](const AddonInstance* addon) {
-        return addon->toAddon->GetBackendName(addon, strBackendName, sizeof(strBackendName));
-      },
-      true, false);
-
-  if (retVal != PVR_ERROR_NO_ERROR)
-    return false;
-
-  /* get the connection string */
-  retVal = DoAddonCall(
-      __func__,
-      [&strConnectionString](const AddonInstance* addon) {
-        return addon->toAddon->GetConnectionString(addon, strConnectionString,
-                                                   sizeof(strConnectionString));
-      },
-      true, false);
-
-  if (retVal != PVR_ERROR_NO_ERROR && retVal != PVR_ERROR_NOT_IMPLEMENTED)
-    return false;
-
-  /* display name = backend name:connection string */
-  strFriendlyName = strlen(strConnectionString) == 0 // connection string is optional
-                        ? strBackendName
-                        : StringUtils::Format("{}:{}", strBackendName, strConnectionString);
-
-  /* backend version number */
-  retVal = DoAddonCall(
-      __func__,
-      [&strBackendVersion](const AddonInstance* addon) {
-        return addon->toAddon->GetBackendVersion(addon, strBackendVersion,
-                                                 sizeof(strBackendVersion));
-      },
-      true, false);
-
-  if (retVal != PVR_ERROR_NO_ERROR)
-    return false;
-
-  /* backend hostname */
-  retVal = DoAddonCall(
-      __func__,
-      [&strBackendHostname](const AddonInstance* addon) {
-        return addon->toAddon->GetBackendHostname(addon, strBackendHostname,
-                                                  sizeof(strBackendHostname));
-      },
-      true, false);
-
-  if (retVal != PVR_ERROR_NO_ERROR && retVal != PVR_ERROR_NOT_IMPLEMENTED)
-    return false;
+  const std::string strFriendlyName = m_strFriendlyName;
```

Comment:
this is problematic as you are accessing the member without mutex protection.

Reply:
Right.